arangod/IResearch/IResearchFeature.cpp

```diff
@@ -310,27 +310,39 @@ bool upgradeArangoSearchLinkCollectionName(TRI_vocbase_t& vocbase,
   // persist collection names in links
   for (auto& collection : vocbase.collections(false)) {
     auto indexes = collection->getIndexes();
-    auto clusterCollection =
-        clusterInfo.getCollectionNT(vocbase.name(),
-                                    collection->name());
-    if (clusterCollection) {
-      auto name = clusterCollection->name();
+    std::string clusterCollectionName;
+    if (!collection->shardIds()->empty()) {
+      unsigned tryCount{60};
+      do {
+        LOG_TOPIC("423b3", TRACE, arangodb::iresearch::TOPIC)
+            << " Checking collection '" << collection->name() << "' in database '" << vocbase.name() << "'";
+        clusterCollectionName = clusterInfo.getCollectionNameForShard(collection->name());
```

Comment:
let's leave a comment of why do we use `getCollectionNameForShard` instead of `getCollectionNT`

Reply:
done